backend/alert/alerts.py

```diff
@@ -48,46 +51,127 @@ def __init__(self, template, reg):
+                }
+            )
         self.registration = reg
 
     @abstractmethod
-    def send_alert(self):
+    def send_alert(self, close_notification=False):
         pass
 
 
 class Email(Alert):
     def __init__(self, reg):
-        super().__init__("alert/email_alert.html", reg)
+        super().__init__(
+            "alert/email_alert.html", reg, close_template="alert/email_close_alert.html"
+        )
 
-    def send_alert(self):
+    def send_alert(self, close_notification=False):
+        """
+        Returns False if notification was not sent intentionally,
+        and None if notification was attempted to be sent but an error occurred.
+        """
         if self.registration.user is not None and self.registration.user.profile.email is not None:
             email = self.registration.user.profile.email
         elif self.registration.email is not None:
             email = self.registration.email
         else:
             return False
+
         try:
+            thread_subject = f"{self.registration.section.full_code} is now open!"
+            if close_notification:
+                if not self.close_text:
+                    # This should be unreachable
+                    return None
+                alert_subject = f"RE: {thread_subject}"
+                alert_text = self.close_text
+            else:
+                alert_subject = thread_subject
+                alert_text = self.text
             return send_email(
                 from_="Penn Course Alert <team@penncoursealert.com>",
                 to=email,
-                subject="%s is now open!" % self.registration.section.full_code,
-                html=self.text,
+                subject=alert_subject,
+                html=alert_text,
             )
         except SMTPRecipientsRefused:
             logger.exception("Email Error")
-            return False
+            return None
 
 
 class Text(Alert):
     def __init__(self, reg):
-        super().__init__("alert/text_alert.txt", reg)
-
-    def send_alert(self):
+        super().__init__("alert/text_alert.txt", reg, close_template="alert/text_alert_close.txt")
+
+    def send_alert(self, close_notification=False):
+        """
+        Returns False if notification was not sent intentionally,
+        and None if notification was attempted to be sent but an error occurred.
+        """
+        if self.registration.user is not None and self.registration.user.profile.push_notifications:
+            # Do not send text if push_notifications is enabled
+            return False
         if self.registration.user is not None and self.registration.user.profile.phone is not None:
             phone_number = self.registration.user.profile.phone
         elif self.registration.phone is not None:
             phone_number = self.registration.phone
         else:
             return False
 
-        return send_text(phone_number, self.text)
+        if close_notification:
```

Comment:
We should never be sending close notifications via text messages -- this should be `return None` whether or not `close_text` is available.